config/src/config/logger_config.rs

```diff
@@ -86,21 +91,96 @@ impl ConfigSanitizer for LoggerConfig {
     }
 }
 
-/// Returns true iff the tokio-console feature is enabled
-fn is_tokio_console_enabled() -> bool {
-    cfg_if! {
-        if #[cfg(feature = "tokio-console")] {
-            true
-        } else {
-            false
+impl ConfigOptimizer for LoggerConfig {
+    fn optimize(
+        node_config: &mut NodeConfig,
+        local_config_yaml: &Value,
+        _node_type: NodeType,
+        _chain_id: ChainId,
+    ) -> Result<bool, Error> {
+        let logger_config = &mut node_config.logger;
+        let local_logger_config_yaml = &local_config_yaml["logger"];
+
+        // Set the tokio console port
+        let mut modified_config = false;
+        if local_logger_config_yaml["tokio_console_port"].is_null() {
+            // If the tokio-console feature is enabled, set the default port.
+            // Otherwise, disable the tokio console port.
+            if is_tokio_console_enabled() {
```

Comment:
What happens if the `tokio_console_port` is already specified in the config? We shouldn't be overriding them right?

Reply:
Yeah, exactly. 😄 The line `if local_logger_config_yaml["tokio_console_port"].is_null()` will ensure that the logic only runs if the `tokio_console_port` is not set in the configs.

Reply:
Ignore my previous comment, I see that you are checking for `null` above.